docs/core/model_training.md

```diff
@@ -0,0 +1,341 @@
+- Initial Structure to Relaxed Energy prediction (IS2RE)
+- Structure to Energy and Forces (S2EF)
+- Initial Structure to Relaxed Structure (IS2RS)
+
+This document is a tutorial for training and evaluating models
+for each of these tasks as well as generating submission files for the
+[evaluation server hosted on EvalAI](https://eval.ai/web/challenges/challenge-page/712/overview).
+
+`main.py` serves as the entry point to run any task. This script requires two command line
+arguments at a minimum:
+* `--mode MODE`: MODE can be `train`, `predict` or `run-relaxations` to train a model, make predictions
+using an existing model, or run machine learning based relaxations using an existing model, respectively.
+* `--config-yml PATH`: PATH is the path to a YAML configuration file. We use YAML files to supply all
+parameters to the script. The `configs` directory contains a number of example config files.
+
+Running `main.py` directly runs the model on a single CPU or GPU if one is available:
+```
+python main.py --mode train --config-yml configs/TASK/SIZE/MODEL/MODEL.yml
+```
+If you have multiple
+GPUs, you can use distributed data parallel training by running:
+```
+python -u -m torch.distributed.launch --nproc_per_node=8 main.py --distributed --num-gpus 8 [...]
+```
+`torch.distributed.launch` launches multiple processes for distributed training. For more details, refer to
+https://pytorch.org/docs/stable/distributed.html#launch-utility
+
+If training with multiple GPUs, GPU load balancing may be used to evenly distribute a batch of variable system sizes across GPUs. Load balancing may either balance by number of atoms or number of neighbors. A `metadata.npz` file must be available in the dataset directory to take advantage of this feature. The following command will generate a  `metadata.npz` file and place it in the corresponding directory.
+```
+python scripts/make_lmdb_sizes.py --data-path data/s2ef/train/2M --num-workers 8
+```
+Load balancing is activated by default (in atoms mode). To change modes you can specify the following in your config:
+```
+optim:
+  load_balancing: neighbors
+```
+For more details, refer to https://github.com/Open-Catalyst-Project/ocp/pull/267.
+
+If you have access to a slurm cluster, we use the [submitit](https://github.com/facebookincubator/submitit) package to simplify multi-node distributed training:
+```
+python main.py --distributed --num-gpus 8 --num-nodes 6 --submit [...]
+```
+
+In the rest of this tutorial, we explain how to train models for each task.
+
+# OC20
+
+## Initial Structure to Relaxed Energy prediction (IS2RE)
+
+In the IS2RE tasks, the model takes the initial structure as an input and predicts the structure’s adsorption energy
+in the relaxed state. To train a model for the IS2RE task, you can use the `EnergyTrainer`
+Trainer and `SinglePointLmdb` dataset by specifying the following in your configuration file:
```

Comment:
Single....>LmdbDataset

Reply:
updated all of these instances, but might still need another pass so that it reads well